master/paths.html

```diff
@@ -1006,78 +1010,138 @@ <h2 id="PathDirectionality">Path directionality</h2>
+  follows.
+</p>
+<p>
+  If the current point is (<var>curx</var>, <var>cury</var>) and the
+  final control point of the previous <a>path segment</a> is
+  (<var>oldx2</var>, <var>oldy2</var>), then the reflected point (i.e., (<var>newx1</var>,
+  <var>newy1</var>), the first control point of the current <a>path
+  segment</a>) is:
+</p>
 <pre>
 (newx1, newy1) = (curx - (oldx2 - curx), cury - (oldy2 - cury))
                = (2*curx - oldx2, 2*cury - oldy2)
 </pre>
-  </li>
 
-  <li>A non-positive radius value is an error.</li>
+<h3 id="ZeroLengthSegments">Zero-length path segments</h3>
 
-  <li>Unrecognized contents within a path data stream (i.e.,
-  contents that are not part of the path data grammar) is an
-  error.</li>
+<p>Path segments with zero length are not invalid,
+and will affect rendering in the following cases:</p>
+
+<ul>
+  <li>If markers are specified, then a marker is drawn on
+  every applicable vertex, even if the given vertex is the
+  end point of a zero-length <a>path segment</a> and even if
+  "moveto" commands follow each other.</li>
+
+  <li>As mentioned in <a href="painting.html#StrokeProperties">Stroke Properties</a>,
+  linecaps must be painted for zero-length subpaths when
+  <a>'stroke-linecap'</a> has a value of
+  <span class='prop-value'>round</span> or
+  <span class='prop-value'>square</span>.</li>
 </ul>
 
+<h3 id="PathDataErrorHandling">Error handling in path data</h3>
+  
+  Unrecognized contents within a path data stream (i.e.,
+  contents that are not part of the path data grammar) is an
+  error.
+  In such a case, the following error-handling rules must be used:
+
+  <ul>
+    <li>The general rule for error handling in path data is
+    that the SVG user agent shall render a <a>'path'</a> element up
+    to (but not including) the path command containing the
+    first error in the path data specification. This will
+    provide a visual clue to the user or developer about
+    where the error might be in the path data specification.
+    This rule will greatly discourage generation of invalid
+    SVG path data.</li>
+
+    <li>If a path data command contains an incorrect set of
+    parameters, then the given path data command is rendered
+    up to and including the last correctly defined <a>path segment</a>,
+    even if that <a>path segment</a> is a sub-component of
+    a compound path data command, such as a "lineto" with
+    several pairs of coordinates. For example, for the path
+    data string <span class='attr-value'>'M 10,10 L 20,20,30'</span>,
+    there is an odd number of parameters for the "L" command, which requires an even
+    number of parameters. The user agent is required to draw
+    the line from (10,10) to (20,20) and then perform error
+    reporting since <span class='attr-value'>'L 20 20'</span>
+    is the last correctly defined segment of the path data specification.</li>
+
+    <li>Wherever possible, all SVG user agents shall report
```

Comment:
Agree, it might be neater to downgrade to a "should" since the "wherever possible" already gives wiggle room.

In the general error processing guidelines (moved to conform.html lines 730+ in the final version of this PR), there's a longer bit about reporting errors to the user, but it is even more vague:

<blockquote>
<li>If the user agent has access to an error reporting
  capability such as status bar or console, it is recommended that the
  user agent provide whatever additional detail it can to
  enable the user or developer to quickly find the source of
  the error. For example, the user agent might provide an error
  message along with a line number and character number at
  which the error was encountered.</li>
</blockquote>

I added in the "or console" bit, otherwise the wording is straight from SVG 1.0.  As is the ["wherever possible, [UAs] shall" wording](https://www.w3.org/TR/SVG10/implnote.html#PathElementImplementationNotes).